### DISCUSSION TITLE
设计好一个选股策略之后，如何通过数据对这个策略的历史表现给出公正、客观的分析？

### DISCUSSION THREAD

**Discussion by zengbin93** [waditu/czsc#34] [Ideas]
有好的想法，欢迎畅所欲言，任何有价值的想法我都会实现并开源到 `czsc`；先感谢各位了！！！

**Comment by zengbin93** [comment]
## 评价方法：1）N周期区间百分位；2）N周期绝对收益

### 评价方法一：N周期区间百分位
假设选股策略确认选股时间为 T，股价为 P，T+N 区间内的K线最高价为 MAX_P，最低价为 MIN_P，则 N周期区间百分位 = (P - MIN_P) / (MAX_P - MIN_P) * 100

### 评价方法二：N周期绝对收益
假设选股策略确认选股时间为 T，股价为 P，第T+N 根K线收盘价为 C，则 N周期绝对收益 = (C - P) / P * 100

**Comment by jhaoq** [comment]
做出来像tradingview 一样的策略回测数据 https://www.tradingview.com/blog/en/new-features-improvements-strategy-backtesting-3643/， 其中比较有代表性的是sharpe比率， 赢率， 和赢钱系数（profit factor）

**Comment by jackyq2015** [comment]
光選股策略不好單獨評價，需要同時考慮賣出策略才能計算勝率，收益等。

**Comment by dhaily** [comment]
就是不单只是选股，还要设置好买卖点，运行一两年的数据，得到夏普率 胜率等数据

**Comment by xiewangsun** [comment]
Walk-Forward Analysis

**Comment by tomjamescn** [comment]
量化交易的那一套评估方法
